ssl/ssl_sess.c

```diff
@@ -713,7 +717,8 @@ int SSL_CTX_add_session(SSL_CTX *ctx, SSL_SESSION *c)
                 if (!remove_session_lock(ctx, ctx->session_cache_tail, 0))
                     break;
                 else
-                    ctx->stats.sess_cache_full++;
+                    CRYPTO_atomic_add(&ctx->stats.sess_cache_full, 1, &discard,
+                                      ctx->lock);
```

Comment:
I'm wondering if it would be useful that CRYPTO_atomic_add() supported NULL for the return value.

Reply:
I'm torn here -- it is a good suggestion but making the user think is also good.

I rather suspect that the API users will discard the value regardless.